src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -45,6 +45,26 @@ namespace OpenMS
 
   PeakIntegrator::~PeakIntegrator() {}
 
+  double PeakIntegrator::estimateBackground(
+    const MSChromatogram& chromatogram,
+    const double& left,
+    const double& right
+  )
+  {
+    const double intensity_l = chromatogram.RTBegin(left)->getIntensity();
+    const double intensity_r = (chromatogram.RTEnd(right)-1)->getIntensity();
+    const double delta_int = intensity_r - intensity_l; // sign will determine line direction
+    const double delta_input_rt = right - left;
+    double background = 0.0;
+    for (auto it=chromatogram.RTBegin(left); it!=chromatogram.RTEnd(right); ++it)
+    {
+      // calculate the background using the formula
+      // y = mx + b where x = retention time, m = slope, b = left intensity
+      background += delta_int / delta_input_rt * (it->getRT() - left) + intensity_l;
```

Comment:
Be sure to add in the other cases as well based on the integration_type.  I believe in the code the uncommented version was for intensity_sum while the commented version was for trapezoid